core/src/main/java/feign/ReflectiveFeign.java

```diff
@@ -57,15 +57,21 @@
   public <T> T newInstance(Target<T> target) {
     Map<String, MethodHandler> nameToHandler = targetToHandlersByName.apply(target);
     Map<Method, MethodHandler> methodToHandler = new LinkedHashMap<Method, MethodHandler>();
+    Promise<Object> proxyPromise = new Promise<Object>();
+
     for (Method method : target.type().getMethods()) {
       if (method.getDeclaringClass() == Object.class) {
         continue;
+      } else if(Util.isDefault(method)) {
+        methodToHandler.put(method, new DefaultMethodHandler(method, proxyPromise));
```

Comment:
great. some touches that might make me be able to merge right away:

* make DefaultMethodHandler package private and final (we don't want
people trying to mutate or extend this)
* update CHANGES with a note about supporting default methods
* put a cool example in README about what this allows you to do! ex
Github api = Github.connect();